packages/kbn-test/src/functional_test_runner/lib/config/schema.ts

```diff
@@ -248,5 +248,20 @@ export const schema = Joi.object()
         fixedHeaderHeight: Joi.number().default(50),
       })
       .default(),
+
+    // settings for the security service if there is no defaultRole defined, then default to superuser role.
+    security: Joi.object()
+      .keys({
+        roles: Joi.object().default(),
+        defaultRoles: Joi.array()
+          .items(Joi.string())
+          .when('$primary', {
```

Comment:
@spalger - can you plz explain this ?

Reply:
`$primary` means that the `primary` key will be looked up on the context object getting passed in, e.g. here https://github.com/elastic/kibana/blob/e3bcd2027d79d279d25151a9fd4fed5be9b5179d/packages/kbn-test/src/functional_test_runner/lib/config/config.ts#L49

`$` is used for looking keys up in the context

Reply:
Thanks Brian for explaining it.

Reply:
In this instance it's an indication that this is the root config, if this is a config that is loaded via `readConfigFile` then primary will be false. We do this so that we can have incomplete config files like `common/config.ts` that doesn't mean the minimum requirements like listing at least one default role.

Reply:
Thanks for the explanation!